File: Triatlon-CS/Persistence/UserDBRepository.cs
using System.Collections.Generic;
using System.Data;
using log4net;
using Triatlon_CSharp.Domain;
using Triatlon_CSharp.Utils;

namespace Triatlon_CSharp.Repository
{
    public class UserDBRepository : UserRepository
    {
        private static readonly ILog Log = LogManager.GetLogger("UserDBRepository");

        public UserDBRepository()
        {
            Log.Info("Creating UserDBRepository");
        }
        
        public User FindOne(int id)
        {
            Log.InfoFormat("Entering FindOne with value {0}", id);
            IDbConnection connection = DBUtils.getConnection();

            User searchedUser = null;

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select * from Users where id=@id";
                var paramId = command.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                command.Parameters.Add(paramId);

                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        int idU = dataReader.GetInt32(0);
                        int idStage = dataReader.GetInt32(1);
                        string firstName = dataReader.GetString(2);
                        string lastName = dataReader.GetString(3);
                        string username = dataReader.GetString(4);
                        string password = dataReader.GetString(5);

                        Stage stage = new Stage();
                        stage.Id = idStage;

                        User user = new User(firstName, lastName, username, password, stage);
                        user.Id = idU;

                        searchedUser = user;
                    }
                }
            }

            return searchedUser;
        }
        
        public User FindByUsername(string username)
        {
            Log.InfoFormat("Entering FindOne with value {0}", username);
            IDbConnection connection = DBUtils.getConnection();

            User searchedUser = null;

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select * from Users where username=@username";
                var paramUsername = command.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = username;
                command.Parameters.Add(paramUsername);

                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        int idU = dataReader.GetInt32(0);
                        int idStage = dataReader.GetInt32(1);
                        string firstName = dataReader.GetString(2);
                        string lastName = dataReader.GetString(3);
                        string username2 = dataReader.GetString(4);
                        string password = dataReader.GetString(5);

                        Stage stage = new Stage();
                        stage.Id = idStage;

                        User user = new User(firstName, lastName, username2, password, stage);
                        user.Id = idU;

                        searchedUser = user;
                    }
                }
            }

            return searchedUser;
        }

        public IEnumerable<User> FindAll()
        {
            throw new System.NotImplementedException();
        }

        public User Save(User entity)
        {
            var connection = DBUtils.getConnection();

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "insert into Users (idStage, firstName, lastName, username, password) values (@idStage, @firstName, @lastName, @username, @password)";

                var paramIdStage = command.CreateParameter();
                paramIdStage.ParameterName = "@idStage";
                paramIdStage.Value = entity.Stage.Id;
                command.Parameters.Add(paramIdStage);
                
                var paramFirstName = command.CreateParameter();
                paramFirstName.ParameterName = "@firstName";
                paramFirstName.Value = entity.FirstName;
                command.Parameters.Add(paramFirstName);
                
                var paramLastName = command.CreateParameter();
                paramLastName.ParameterName = "@lastName";
                paramLastName.Value = entity.LastName;
                command.Parameters.Add(paramLastName);
                
                var paramUsername = command.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = entity.Username;
                command.Parameters.Add(paramUsername);
                
                var paramPassword = command.CreateParameter();
                paramPassword.ParameterName = "@password";
                paramPassword.Value = entity.Password;
                command.Parameters.Add(paramPassword);

                var result = command.ExecuteNonQuery();
            }

            return null;
        }

        public int verifyAccount(string username, string password)
        {
            Log.InfoFormat("Entering verifyAccount with value {0}", username);
            IDbConnection connection = DBUtils.getConnection();

            int id = -1;

            using (var command = connection.CreateCommand())
            {
                command.CommandText = "select id from Users where username=@username and password=@password";
                var paramUsername = command.CreateParameter();
                paramUsername.ParameterName = "@username";
                paramUsername.Value = username;
                command.Parameters.Add(paramUsername);

                var paramPassword = command.CreateParameter();
                paramPassword.ParameterName = "@password";
                paramPassword.Value = password;
                command.Parameters.Add(paramPassword);

                using (var dataReader = command.ExecuteReader())
                {
                    if (dataReader.Read())
                    {
                        id = dataReader.GetInt32(0);
                    }
                }
            }

            return id;
        }

        public User Delete(int id)
        {
            throw new System.NotImplementedException();
        }

        public User Update(User entity)
        {
            throw new System.NotImplementedException();
        }
    }
}

File: Triatlon-Java/ChatPersistence/src/main/java/triathlon/persistence/repository/jdbc/triatlon/UserDBRepository.java
package triathlon.persistence.repository.jdbc.triatlon;

import triathlon.model.Stage;
import triathlon.model.User;


import triathlon.persistence.UserRepository;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.Properties;

public class UserDBRepository implements UserRepository {
    private JdbcUtils dbUtils;
    private static final Logger logger = LogManager.getLogger();

    public UserDBRepository(Properties properties) {
        logger.info("Initializing UserDBRepository with properties: {}", properties);
        this.dbUtils = new JdbcUtils(properties);
    }

    @Override
    public User findOne(Integer integer) {
        User searchedUser = null;
        logger.traceEntry();
        Connection connection = dbUtils.getConnection();

        try (PreparedStatement preparedStatement = connection.prepareStatement("select * from Users where id = ?")){
            preparedStatement.setInt(1, integer);
            try (ResultSet resultSet = preparedStatement.executeQuery()){
                while (resultSet.next()) {
                    int id = resultSet.getInt("id");
                    int idStage = resultSet.getInt("idStage");
                    String firstName = resultSet.getString("firstName");
                    String lastName = resultSet.getString("lastName");
                    String username = resultSet.getString("username");
                    String password = resultSet.getString("password");

                    Stage stage = new Stage();
                    stage.setId(idStage);

                    User user = new User(firstName, lastName, username, password, stage);
                    user.setId(id);
                    searchedUser = user;
                }
            }
        }
        catch (SQLException ex) {
            logger.error(ex);
            System.err.println(ex);
        }

        return searchedUser;
    }

    @Override
    public Iterable<User> findAll() {
        return null;
    }

    @Override
    public User save(User entity) {
        return null;
    }

    @Override
    public User delete(Integer integer) {
        return null;
    }

    @Override
    public User update(User entity) {
        return null;
    }

    @Override
    public Integer verifyAccount(String username, String password) {
        logger.traceEntry();
        Connection connection = dbUtils.getConnection();
        Integer id = -1;

        try (PreparedStatement preparedStatement = connection.prepareStatement("select id from Users where username=? and password=?")){
            preparedStatement.setString(1, username);
            preparedStatement.setString(2, password);
            try (ResultSet resultSet = preparedStatement.executeQuery()){
                while (resultSet.next()) {
                    id = resultSet.getInt("id");
                }
            }
        }
        catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }

        return id;
    }
}
